baselines/depthfl/README.md

```diff
@@ -0,0 +1,170 @@
+
+```bash
+# table 2 (HeteroFL row)
+python -m depthfl.main --config-name="heterofl" 
+python -m depthfl.main --config-name="heterofl" --multirun exclusive_learning=true model.scale=false model_size=1,2,3,4 
+
+# table 2 (DepthFL(FedAvg) row)
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false 
+python -m depthfl.main --multirun fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=1,2,3,4
+
+# table 2 (DepthFL row)
+python -m depthfl.main
+python -m depthfl.main --multirun exclusive_learning=true model_size=1,2,3,4
+```
+
+**Table 2** 
+
+100% (a), 75%(b), 50%(c), 25% (d) cases are exclusive learning scenario. 100% (a) exclusive learning means, the global model and every local model are equal to the smallest local model, and 100% clients participate in learning. Likewise, 25% (d) exclusive learning means, the global model and every local model are equal to the larget local model, and only 25% clients participate in learning.
+
+| Scaling Method | Dataset | Global Model | 100% (a) | 75% (b) | 50% (c) | 25% (d) | 
+| :---: | :---: | :---: | :---: | :---: | :---: | :---: | 
+| HeteroFL<br>DepthFL (FedAvg)<br>DepthFL | CIFAR100 | 57.61<br>72.67<br>76.06 | 64.39<br>67.08<br>69.68 | 66.08<br>70.78<br>73.21 | 62.03<br>68.41<br>70.29 | 51.99<br>59.17<br>60.32 |
+
+```bash
+# table 3 (Width Scaling - Duplicate results from table 2)
+python -m depthfl.main --config-name="heterofl" 
+python -m depthfl.main --config-name="heterofl" --multirun exclusive_learning=true model.scale=false model_size=1,2,3,4 
+
+# table 3 (Depth Scaling : Exclusive Learning, DepthFL(FedAvg) rows - Duplicate results from table 2)
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false 
+python -m depthfl.main --multirun fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=1,2,3,4
+
+## table 3 (Depth Scaling - InclusiveFL row)
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false
+```
+
+**Table 3** 
+
+Accuracy of global sub-models compared to exclusive learning on CIFAR-100.
+
+| Method | Algorithm | Classifier 1/4 | Classifier 2/4 | Classifier 3/4 | Classifier 4/4 |
+| :---: | :---: | :---: | :---: | :---: | :---: |
+| Width Scaling | Exclusive Learning<br>HeteroFL| 64.39<br>51.08 | 66.08<br>55.89 | 62.03<br>58.29 | 51.99<br>57.61 |
+
+| Method | Algorithm | Classifier 1/4 | Classifier 2/4 | Classifier 3/4 | Classifier 4/4 |
+| :---: | :---: | :---: | :---: | :---: | :---: |
+| Depth Scaling | Exclusive Learning<br>InclusiveFL<br>DepthFL (FedAvg) | 67.08<br>47.61<br>66.18 | 68.00<br>53.88<br>67.56 | 66.19<br>59.48<br>67.97 | 56.78<br>60.46<br>68.01 |
+
+```bash
+# table 4
+python -m depthfl.main --multirun fit_config.kd=true,false dataset_config.iid=true,false
```

Comment:
Is this command enough to generate the whole Table 4? I can see how it will generate  columns `Classifier 1/4 | Classifier 2/4 | Classifier 3/4 | Classifier 4/4` but I'm not so sure about the `Ensemble` column. Or is it obvious how to obtain it when running the experiments? (I have only run Table 2 and I was able to reproduce your results)

Reply:
The ensemble column refers to the ensemble model of four classifiers. When running an experiment, 'accuracy' in log refers to the accuracy of the ensemble model, and 'accuracy_single' refers to the performance of each classifier.

Reply:
Ok. Do you think it would be useful to add the second sentence you wrote somewhere in the `Running the Experiments` section in the `README.md`?

Reply:
I added this comment to ```Running the Experiments``` section.